telemetry/csharp/ToolkitTelemetryGenerator/Program.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using CommandLine;
+using ToolkitTelemetryGenerator.Models;
+
+namespace ToolkitTelemetryGenerator
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Options options = null;
+            Parser.Default.ParseArguments<Options>(args)
+                .WithParsed<Options>(o =>
+                {
+                    if (string.IsNullOrWhiteSpace(o.OutputFolder))
+                    {
+                        o.OutputFolder = Directory.GetCurrentDirectory();
+                    }
+
+                    options = o;
+                });
+
+            if (options == null)
+            {
+                throw new Exception("Program Options are undefined or missing.");
+            }
+
+            var definitionPath = Path.Combine(GetTelemetryDefinitionsFolder(), "commonDefinitions.json");
+
+            var definitions = TelemetryDefinitions.Load(definitionPath);
+
+            DefinitionsBuilder builder = new DefinitionsBuilder()
+                .WithNamespace(options.Namespace)
+                .AddMetrics(definitions.metrics)
+                .AddMetricsTypes(definitions.types);
+
+            var code = builder.Build();
+
+            File.WriteAllText(Path.Combine(options.OutputFolder, "GeneratedCode.cs"), code);
```

Comment:
GeneratedCode.cs? shouldn't it have a real name?

Reply:
In the current form (where you have to place the file somewhere in the VS Toolkit), the filename is arbitrary. In a future iteration where we produce a full package, the filename is irrelevant.